crates/node/src/tee/allowed_image_hashes_watcher.rs

```diff
@@ -371,15 +412,106 @@ mod tests {
+        let allowed_images = vec![image_hash_1(), image_hash_2(), image_hash_3()];
+        let full_list = allowed_images.clone();
+
+        // Current running image (could be any)
+        let current_image = image_hash_1();
+
+        let cancellation_token = CancellationToken::new();
+        let (_sender, receiver) = watch::channel(allowed_images.clone());
+        let (sender_shutdown, mut receiver_shutdown) = mpsc::channel(1);
+
+        let write_is_called = Arc::new(Notify::new());
+
+        // Mock storage expecting exactly the full list
+        let mut storage_mock = MockAllowedImageHashesStorage::new();
+        {
+            let expected = full_list.clone();
+            let write_is_called = write_is_called.clone();
+
+            storage_mock
+                .expect_set()
+                .once()
+                .with(predicate::eq(expected))
+                .returning(move |_| {
+                    write_is_called.notify_one();
+                    Box::pin(async { Ok(()) })
+                });
+        }
+
+        let _join_handle = tokio::spawn(monitor_allowed_image_hashes(
+            cancellation_token,
+            current_image,
+            receiver,
+            storage_mock,
+            sender_shutdown,
+        ));
+
+        // Wait for the write
+        write_is_called.notified().await;
+
+        // No shutdown expected
+        assert_matches!(
+            receiver_shutdown.try_recv(),
+            Err(TryRecvError::Empty),
+            "Shutdown signal should not be sent"
         );
     }
+
+    /// Verifies that an empty allowed-hash list from the contract is handled safely:
+    /// the watcher must not write to storage, must not send a shutdown signal,
+    /// and must continue running without panicking.
+    #[tokio::test]
+    async fn test_empty_hash_list_does_not_panic() {
+        // Contract sends an empty list
+        let allowed_images: Vec<MpcDockerImageHash> = vec![];
+        let current_image = image_hash_1();
+
+        let cancellation_token = CancellationToken::new();
+        let (_sender, receiver) = watch::channel(allowed_images);
+        let (sender_shutdown, mut receiver_shutdown) = mpsc::channel(1);
+
+        // Storage must NOT be called
+        let mut storage_mock = MockAllowedImageHashesStorage::new();
+        storage_mock.expect_set().never();
+
+        // Spawn watcher
+        let _join_handle = tokio::spawn(monitor_allowed_image_hashes(
+            cancellation_token.child_token(),
+            current_image,
+            receiver,
+            storage_mock,
+            sender_shutdown,
+        ));
+
+        // No shutdown signal should be sent
+        assert_matches!(
+            receiver_shutdown.try_recv(),
+            Err(TryRecvError::Empty),
+            "Shutdown should NOT be sent when list is empty"
+        );
+    }
+    #[test]
+    fn test_json_key_matches_launcher() {
+        assert_eq!(JSON_KEY_APPROVED_HASHES, "approved_hashes");
+    }
```

Comment:
same as in pytests, I think this test does not add any value. The comment above the constant already does this. But feel free to disregard this, anyway the test is not expensive :)